### ISSUE TITLE
Bluetooth Connection after command lost

### ISSUE THREAD

**Issue by heradon** [derkalle4/python3-idotmatrix-client#9]
Hello guys. 

First, thank you for your work! AWESOME!
But my problem is, after a successful command i cannot send a second command to the Pixel Display.
i must restart my bluetooth daemon (systemctl restart bluetooth) then i can send the second command.

My System:

SYS: ArchLinux
Kernel: 6.6.8-arch1
HW: Asrock x300

Greetz Hera

**Comment by derkalle4** [comment]
Hi Heradon,

many thanks for your bug report!

This is unfortunately my first project with utilizing Bluetooth with Python and somehow Bleak does not work that good (or it has something to do with the display itself that makes it worse).

I have a quite similar problem as well: I need to send the first command twice to be able to send a successful command to the device.

My further plans would be to initiate the connection to the device one time and have the ability to send the next command using the already opened connection. Currently this is not possible. Thought I would integrate a better command line solution as well as a gui.

Keeping this bug open and assign it to myself. Hope to be able to provide you a solution.

Kind Regards,
Kalle

**Comment by derkalle4** [comment]
Hi @heradon,

are you familiar with Python? Then you could check out the early version of the library. Will be available via pip soon: https://github.com/derkalle4/python3-idotmatrix-library

There is a test.py which should work, this will send multiple command over the opened connection.

**Comment by derkalle4** [comment]
Hi @heradon,

I have implemented the library into this project. Could you check again using the --test parameter? This should send multiple commands without reconnecting! It works fine on Linux Mint 22.10.

Example command:

./run_in_venv.sh --address auto --test

Kind Regards,
Kalle

**Comment by aviwad** [comment]
The test works! Can this be integrated for regular commands? Would love to update images on my display without the display symbol for reconnection.

**Comment by derkalle4** [comment]
@aviwad 

yeah, there is a planned integration with some kind of open connection. Unfortunately when the python script closes after a command got sent there is no way to keep the connection open (I guess). Maybe the bluetooth library I use can tell the operating system to keep the connection open so we can re-use it.
